File: domain/src/main/java/com/tonitealive/app/domain/interactors/UseCase.java
package com.tonitealive.app.domain.interactors;

import com.tonitealive.app.domain.executor.PostExecutionThread;
import com.tonitealive.app.domain.executor.ThreadExecutor;


public abstract class UseCase {

    protected final ThreadExecutor executor;
    protected final PostExecutionThread postExecutionThread;

    protected UseCase(ThreadExecutor executor, PostExecutionThread postExecutionThread) {
        this.executor = executor;
        this.postExecutionThread = postExecutionThread;
    }

}

File: domain/src/main/kotlin/com/tonitealive/app/domain/interactors/UseCase.kt
package com.tonitealive.app.domain.interactors

import com.tonitealive.app.domain.executor.PostExecutionThread
import com.tonitealive.app.domain.executor.ThreadExecutor


abstract class UseCase protected constructor(val executor: ThreadExecutor,
                                             val postExecutionThread: PostExecutionThread) {

}